services/base-jdbc/src/main/java/org/eclipse/hono/service/base/jdbc/store/device/TableManagementStore.java

```diff
@@ -926,25 +958,41 @@ private List<CommonCredential> parseCredentials(final ResultSet result) {
      * @param pageSize The page size.
      * @param pageOffset The page offset.
      * @param filters The list of filters (currently only the first value of the list is used).
+     * @param isGateway Optional search gateway or only devices filter.
      * @param spanContext The span to contribute to.
      * @return A future containing devices.
      */
-    public Future<SearchResult<DeviceWithId>> findDevices(final String tenantId, final int pageSize, final int pageOffset, final List<Filter> filters,
+    public Future<SearchResult<DeviceWithId>> findDevices(final String tenantId, final int pageSize, final int pageOffset, final List<Filter> filters, final Optional<Boolean> isGateway,
                                                           final SpanContext spanContext) {
 
 
-        final var filter = filters.stream().findFirst();
+        final Statement findDeviceSqlStatement;
+        final Statement countStatement;
+        final String field;
+        final String value;
+
+        if (isGateway.isPresent()) {
```

Comment:
so, my understanding is that I can  search using *either* filters *or* the `isGateway` predicate, right? But I cannot search non-gateway devices having a particular property value?

Reply:
Yes, that is correct.

Reply:
Still neither the JavaDoc nor device-registry-v1.yaml mention that you can either use filter or isGateway, but not both ...